src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -358,6 +358,14 @@ def _add_event(event_name, properties, instrumentation_key=DEFAULT_INSTRUMENTATI
     })
 
 
+@decorators.suppress_all_exceptions()
+def is_telemetry_enabled():
+    from azure.cli.core.cloud import AZURE_PUBLIC_CLOUD, CLOUDS_FORBIDDING_TELEMETRY
+    if _get_config().get('cloud', 'name', fallback=AZURE_PUBLIC_CLOUD.name) in CLOUDS_FORBIDDING_TELEMETRY:
+        return False
+    return _get_config().getboolean('core', 'collect_telemetry', fallback=True)
```

Comment:
just curious, is the `collect_telemetry` configurable by user?

Reply:
Yes. Users can edit the `~/.azure/config` file to change `collect_telemetry` value in addtion to setting it interactively with `az configure`.